### ISSUE TITLE
Oracle.EntityFrameworkCore issue with Include and same property names

### ISSUE THREAD

**Issue by mbendtsen** [oracle/dotnet-db-samples#132]
There is an issue in Oracle.EntityFrameworkCore v3.19.80 when using `Include()` and the two models have a property of the same name.

Not sure if this is the right place to report this bug. 

### Model
```
public class ModelConfiguration
{
  public int ModelConfigurationId { get; set; }
  public int VendorModelId { get; set; }

  public virtual VendorModel VendorModel { get; set; }
}

public class VendorModel
{
  public int VendorModelId { get; set; }
}
```

### LINQ
```
var modelConfigurations = await modelContext.Modelconfigurations
     .Include(x => x.Vendormodel)
     .FirstOrDefaultAsync();
```

### Generated SQL
```
SELECT
"m"."MODELCONFIGURATIONID",
"m"."VENDORMODELID", 
"v"."VENDORMODELID"
FROM "MODELCONFIGURATIONS" "m"
INNER JOIN "VENDORMODELS" "v" ON "m"."VENDORMODELID" = "v"."VENDORMODELID"
FETCH FIRST 1 ROWS ONLY
```

### Error
```
00918. 00000 -  "column ambiguously defined"
*Cause:    
*Action:
```

If you remove `FETCH FIRST 1 ROWS ONLY` then it works.

**Comment by alexkeh** [comment]
I tested your model against a 19c DB, but didn't encounter the error. Are you using DB version 12.1? If so, you may be hitting the known DB 12.1 issue that triggers the ORA-00918.

The fix is available in DB 12.2 or higher. No need to update the provider if you are hitting this scenario.

**Comment by mbendtsen** [comment]
Yes, I'm using 12.1 (Oracle Database 12c Standard Edition Release 12.1.0.2.0 - 64bit Production)

**Comment by mbendtsen** [comment]
@alexkeh I was a little too fast when writing the issue. I have updated with new information. 
The issue is only there when using `FirstOrDefault()`.

Still thinks that it should alias all columns to avoid situations like this.

**Comment by alexkeh** [comment]
The presence of the row limiting clause with the use of FirstOrDefault() and 12.1 DB fits this known issue symptoms exactly. 

The SQL generated is valid and accurate. The bug (29686264) requires a DB level fix, which was made in DB 12.2, but not backported to 12.1.
